### ISSUE TITLE
Adding known_grads parameter to get_updates method

### ISSUE THREAD

**Issue by nmercado1986** [keras-team/keras#9077]
Sometimes one needs to get the optimizer updates providing evaluation points (also grad_ys in Tensorflow). One can call either T.grad or tf.gradients by importing the backend framework.
This presents an issue, though, as one may need to get the updates to apply those gradients using a specific optimizer.
However, Keras optimizers do not provide a parameter to supply these gradients. If using tensorflow this is not a problem as Tensorflow does provide an opt.apply_gradients function, with which one can do without `opt.get_updates`,  but Theano does not so one ends up implementing the updates oneself. For example, by doing:

`upds = [ (w, w - g * model.optimizer.lr) for w, g in zip(m2.trainable_weights, grads) ]`

In the case of optimizers like RMSprop or Adagrad this becomes extremely tedious and error-prone.
Is there a possibility to support `known_grads` and `grad_ys` to allow for backend-agnostic implementations?
